src/Illuminate/Filesystem/Filesystem.php

```diff
@@ -401,13 +401,11 @@ public function makeDirectory($path, $mode = 0755, $recursive = false, $force =
     public function moveDirectory($from, $to, $overwrite = false)
     {
         if ($overwrite && $this->isDirectory($to)) {
-            $this->deleteDirectory($to);
-
-            $this->copyDirectory($from, $to);
-
-            $this->deleteDirectory($from);
+            if (! $this->deleteDirectory($to)) {
+                return false;
+            }
 
-            return true;
+            return @rename($from, $to) === true;
```

Comment:
Why don't we omit this line? Since the following statement would do the same. Plus we avoid duplication, I think.